File: examples/rust/examples/read_delta_table.rs
//! Read a Delta table with delta-rs. STUB — harness-ready, not yet built out.
//!
//! The region markers and the seed one-liner are in place so the docs can
//! reference this file and `cargo build --examples` compiles it today. The body
//! is filled in (with `deltalake::open_table` + a time-travel read) when the Rust
//! examples are built out.

fn main() -> Result<(), Box<dyn std::error::Error>> {
    // docs-read-delta-table-start
    // let path = docs_factory_seed::seed_dataset("orders")?;
    // let table = deltalake::open_table(&path).await?;
    // println!("{} files", table.get_files_count());
    println!("delta-rs read_delta_table example: not implemented yet");
    // docs-read-delta-table-end
    let _ = docs_factory_seed::seed_dataset; // keep the seed crate linked
    Ok(())
}

File: examples/polars/read_delta_table.py
"""Read a Delta table with Polars. STUB — not yet built out or tested in CI.

The region markers and the seed one-liner are in place so the docs can reference
this file and the test harness can pick it up once implemented. Polars reuses the
Python ``docs_factory_seed`` helper to materialize the table.
"""


def read_delta_table() -> None:
    # --8<-- [start:read-delta-table]
    import polars as pl
    from docs_factory_seed import seed_dataset

    path = seed_dataset("orders")
    df = pl.read_delta(path)
    print(df.head())
    # --8<-- [end:read-delta-table]


if __name__ == "__main__":
    read_delta_table()
